### PR TITLE
Add transformers tool test to pipeline

### PR DESCRIPTION
**Pull request by wangyems**
**Description**: Describe your changes.

1. Setup a pipeline for transformers tool test in Linux CPU machine(an extra <1000s used on Linux CPU pipeline)
2. Add --enable_transformers_tool_test in build.py so that we could only enable it in target machines
3. pytorch1.4 related tests are removed
4. add back some onnx files
5. failed tests are temporarily disabled. Plan to investigate and resolve in future PRs.
7. torch/transformers..package version is hardcoded to avoid unexpected pipeline breaking

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/test/python/transformers/test_attention_fusion.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import os
+import sys
+import onnx
+from bert_model_generator import create_bert_attention, create_tf2onnx_attention_3d
+
+# set path so that we could import from parent directory
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from onnxruntime.transformers.optimizer import optimize_model
+
+# class TestFusion(unittest.TestCase):
```
Comment (tianleiwu) [inline onnxruntime/test/python/transformers/test_attention_fusion.py:17]: Please enable fusion tests

File: onnxruntime/test/python/transformers/test_shape_infer_helper.py [hunk 2]
```diff
@@ -22,25 +23,22 @@ def _load_onnx(self, model_name):
...
         self.assertEqual(shape_infer_helper.get_edge_shape("encoder.layer.4.intermediate.dense.bias"), [3072])
-        self.assertEqual(shape_infer_helper.get_edge_shape("1750"), [3072, 768])
-        self.assertEqual(shape_infer_helper.get_edge_shape("853"), [3])
-        self.assertEqual(shape_infer_helper.get_edge_shape("858"), [1])
-        self.assertEqual(shape_infer_helper.get_edge_shape("880"), [4, 16, 12, 64])
-
-        self.assertEqual(shape_infer_helper.compare_shape("329", "253"), True)
-        self.assertEqual(shape_infer_helper.compare_shape("447", "371"), True)
-        self.assertEqual(shape_infer_helper.compare_shape("329", "817"), False)
+        self.assertEqual(shape_infer_helper.get_edge_shape("880"), [4, 12, 16, 16])
```
Comment (tianleiwu) [inline onnxruntime/test/python/transformers/test_shape_infer_helper.py:37]: why the expected shape changed from [4, 16, 12, 64] to [4, 12, 16, 16]? I think the last dimension shall be 64. Is it a bug in recent shape inference change?

Comment (wangyems) [reply onnxruntime/test/python/transformers/test_shape_infer_helper.py:37]: The graph changed due to transformers update. So the test cases are regenerated

File: onnxruntime/test/python/transformers/test_optimizer.py [hunk 3]
```diff
@@ -331,43 +255,44 @@ def test_huggingface_flaubert_fusion(self):
...
     def test_huggingface_bart_fusion(self):
         self._test_optimizer_on_huggingface_model("facebook/bart-base", [0, 0, 0, 0, 12, 2, 30])
 
-    @pytest.mark.slow
-    def test_huggingface_bert_base_cased_from_tf2onnx(self):
-        self._test_optimizer_on_tf_model("bert-base-cased", [0, 12, 0, 0, 0, 0, 25], 1)
-        self._test_optimizer_on_tf_model("bert-base-cased", [0, 12, 0, 0, 0, 0, 25], 2)
-        self._test_optimizer_on_tf_model("bert-base-cased", [0, 12, 0, 0, 0, 0, 25], 3)
+    # @pytest.mark.slow
+    # def test_huggingface_bert_base_cased_from_tf2onnx(self):
```
Comment (tianleiwu) [inline onnxruntime/test/python/transformers/test_optimizer.py:267]: please enable this test for tf2onnx

Comment (wangyems) [reply onnxruntime/test/python/transformers/test_optimizer.py]: enabled

**Review by tianleiwu — APPROVED** [review]

File: tools/ci_build/build.py [hunk 4]
```diff
@@ -1507,6 +1511,12 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
                 if not args.disable_contrib_ops:
                     run_subprocess([sys.executable, '-m', 'unittest', 'discover', '-s', 'quantization'],
                                    cwd=cwd, dll_path=dll_path)
+                    if args.enable_transformers_tool_test:
+                        required = {
+                            'numpy==1.19.2', 'coloredlogs==15.0', 'tf2onnx==1.8.5', 'transformers==4.6.1',
+                            'torch==1.8.1', 'tensorflow==2.5.0', 'onnxconverter-common==1.8.1', 'psutil'}
```
Comment (tianleiwu) [inline tools/ci_build/build.py:1517]: Since slow tests will not run in pipeline, tf2onnx and tensorflow might be not required.

Comment (tianleiwu) [inline tools/ci_build/build.py:1517]: tf2onnx and tensorflow might not be required since slow tests will not run in pipeline.

**Pull request merged**